### PR TITLE
NAS-110516 / 22.12 / NAS-110516: Add 'retention' to the details-row at snaphots page

### PR DESCRIPTION
**Pull request by bvasilenko**
**Testing**

On **Snapshots** table, new column "Retention" should be displayed at expanded details panel

### CODE REVIEW

File: src/app/interfaces/zfs-snapshot.interface.ts [hunk 1]
```diff
@@ -10,7 +10,11 @@ export interface ZfsSnapshot {
     [property: string]: ZfsProperty<string | number | boolean | ApiTimestamp>;
     creation: ZfsProperty<ApiTimestamp>;
   };
-  retention: unknown;
+  retention?: {
+    datetime: ApiTimestamp;
+    source: string;
```
Comment (undsoft) [inline src/app/interfaces/zfs-snapshot.interface.ts:15]: Enum.

**Review by undsoft — CHANGES_REQUESTED** [review]
Looks good, but I think we can improve messaging here. Seems there different states for this field and we should show this to the user:
* Will be automatically destroyed at datetime by periodic snapshot task
* Is planned to be automatically destroyed at datetime.
* "Will not be destroyed automatically" for null values.

**Comment by bvasilenko** [comment]
@undsoft Now it has enum and improved wordings

**Review by undsoft — APPROVED** [review]
Looks good

**Pull request merged**